test/heavymock/import_records_impl.go

```diff
@@ -24,30 +29,52 @@ func NewHeavymockImporter() *ImporterServer {
 func (s *ImporterServer) Import(stream HeavymockImporter_ImportServer) error {
 	s.mux.Lock()
 	defer s.mux.Unlock()
-	received := make([]exporter.Record, 0)
+	received := make([]*exporter.Record, 0)
 	for {
 		record, err := stream.Recv()
 		if err == io.EOF {
-			s.savedRecords = received
+			s.appendNewRecords(received)
 			return stream.SendAndClose(&Ok{
 				Ok: true,
 			})
 		}
 		if err != nil {
 			return err
 		}
-		received = append(received, *record)
+		received = append(received, record)
 	}
 }
 
-func (s *ImporterServer) GetSavedRecords() []exporter.Record {
+func (s *ImporterServer) GetUnsentRecords() []*exporter.Record {
 	s.mux.Lock()
 	defer s.mux.Unlock()
-	return s.savedRecords
+	res := make([]*exporter.Record, 0)
+	for _, r := range s.records {
+		if !r.isSent {
+			res = append(res, r.record)
+		}
+	}
+
+	return res
 }
 
-func (s *ImporterServer) Cleanup() {
+func (s *ImporterServer) MarkAsSent(records []*exporter.Record) {
 	s.mux.Lock()
 	defer s.mux.Unlock()
-	s.savedRecords = make([]exporter.Record, 0)
+
+	for _, r := range records {
+		for _, s := range s.records {
+			if r.Equal(s.record) {
+				s.isSent = true
+			}
+		}
+	}
+}
+
+func (s *ImporterServer) appendNewRecords(records []*exporter.Record) {
+	slice := make([]*savedRecord, 0)
+	for _, r := range records {
+		slice = append(slice, &savedRecord{r, false})
```

Comment:
no need new slice here if you will add that to s.records  later

Reply:
the point is to add the array of records eventually, not one by one, because this operation is performed while BE is constantly requests for new records (taken from `s.records`)